doc: clearify that http does chunked encoding itself. (#28379)
nodejs/node: doc/api/http.md

* `callback` {Function}
* Returns: {boolean}

This method can be called multiple times. If no `Content-Length` is set, data
will automatically be encoded in HTTP Chunked transfer encoding, so that
server knows when data ends. The `Transfer-Encoding: chunked` header will be
added. Call `end()` when finish sending data so that node would send the

> trailer.

Reviewer: > If no `Content-Length` is set, data will automatically be encoded in HTTP 
Chunked transfer encoding, so that server knows when data ends. The `Transfer-Encoding: chunked` header will be automatically added.

Would be simpler maybe?

Reviewer: With a reference to https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback? Even though it is not a `Writable` the intention is to act as similar as possible. @mcollina, what do you think? Would such a reference be harmful?

`['Transfer-Encoding', 'chunked']` header line when
creating the request.

> Note that this module automatically encode data into chunked encoding.

Reviewer 2: ```suggestion
This module automatically encodes data in chunked encoding.
```

(That's a grammar fix and a style fix, but the content is still problematic, I think.)

* `callback` {Function}
* Returns: {boolean}

> This method can be called multiple times. If no `Content-Length` is set, data

Reviewer 3: I would retain the first sentence

> Sends a chunk of the body.

* `callback` {Function}
* Returns: {boolean}

Sends a chunk of the body. This method can be called multiple times. If no
`Content-Length` is set, data will automatically be encoded in HTTP Chunked
transfer encoding, so that server knows when the data ends. The

> `Transfer-Encoding: chunked` header will be added. Call `end()` when finish

Reviewer 4: Prefer present tense.
```suggestion
`Transfer-Encoding: chunked` header is added. Call `end()` when finish
```

> `Transfer-Encoding: chunked` header will be added. Call `end()` when finish
> sending data so that node would send the trailer.

Reviewer 4: I think the last sentence should be rephrased (does `node` here refers to Node.js? what is a trailer in this context?), and the reference to `end()` should be a link to its documentation.

```suggestion
`Transfer-Encoding: chunked` header will be added. Calling [`request.end()`][]
is necessary to finish sending the request.
```
